feat: add PCI Blueprint (#159)
Basis-Theory/basistheory-integration-guides: collections/blueprints/pci.md

Copy the API Key to be used in the next step.

### Initialize Basis Theory Elements

Let's create a new Card Form component. In a terminal in the same directory as your `App` component, type:

```bash
touch card-form.js
```

In your editor, update your App component with the following code, replacing `PUBLIC_API_KEY` with the API Key from the previous step:

```jsx
import { 
  useBasisTheory,
  BasisTheoryProvider
} from '@basis-theory/basis-theory-react';
import { CardForm } from './card-form';

export default function App() {
  const { bt } = useBasisTheory('PUBLIC_API_KEY', { elements: true });

  return (
    <BasisTheoryProvider bt={bt}>
      <CardForm />
    </BasisTheoryProvider>
  );
}
```

Next, add the following code to your new Card Form component:

```jsx
import { 
  useBasisTheory
} from '@basis-theory/basis-theory-react';

export const CardForm = () => {
  const { bt } = useBasisTheory();

  const submit = async () => {};

  return (
    <form onSubmit={submit} className="form">
      <button type="button" onClick={submit} >
        Submit
      </button>
    </form>
  );
}
```

This will initialize Basis Theory JS and add your new card form.

### Add Card Element

> Now, we need to add a [`CardElement`](https://docs.basistheory.com/elements/#element-types-card-element) component to our form. This Element type offers a single line to capture the card number, expiration date, and CVC.

Reviewer: ```suggestion
Now, we need to add a [`CardElement`](https://docs.basistheory.com/elements/#element-types-card-element) component to our form. This Element type renders a single line containing input fields to capture the card number, expiration date, and CVC.
```

  const submit = async () => {};

  return (
    <form onSubmit={submit} className="form">
      <CardElement id="card" />

      <button type="button" onClick={submit} >
        Submit
      </button>
    </form>
  );
}
```

You can fully customize the [look and feel](./guides/secure/../../../guides/secure/style-elements-for-my-brand.md) to match your existing site and even listen to common [events](https://docs.basistheory.com/elements/#element-events) on all Elements components.

## Store Cards

Now that we have securely captured the card information, we need to securely store the card. 

To do this, we will tokenize the data with Basis Theory. Basis Theory handles all of the secure encryption and storage of the cardholder data and returns a non-sensitive token identifier that can be stored in your database.

### Tokenize the Card

Inside of your `CardForm`, we will update the `submit` function to tokenize the card using the PCI-compliant [`card` Token Type](https://docs.basistheory.com/#token-types-card):

```jsx
const submit = async () => {
  // Tokenize card with Basis Theory
  const token = await bt.tokenize({
    type: 'card',
    data: bt.getElement('card'),
  });

>   // Submit card token to API

Reviewer: Nitpick, but I think we should clarify that this is our app's API, not the BT API

```suggestion
  // Submit card token to our Next.js app's API
```

  // Example using Axios

>   const { data } = await axios.post('/api/cards', { cardTokenId: token.id });

Author: Nah, I meant to replicate posting the tokens to your backend.

>   const { data } = await axios.post('/api/cards', { cardTokenId: token.id });

Author: Updated comment to add more clarity.

};
```

### What is Happening?

> When a user submits their payment information, we will tokenize the underlying value of the [CardElement](https://docs.basistheory.com/elements/#cardelement). This will instruct Basis Theory Elements to submit directly to [Basis Theory's Tokenize endpoint](https://docs.basistheory.com/#tokenize) and return the resulting token identifiers.

Reviewer: ```suggestion
When a user submits their payment information, we will tokenize the underlying value of the [CardElement](https://docs.basistheory.com/elements/#cardelement). This will instruct Basis Theory Elements to submit this sensitive card data directly to [Basis Theory's Tokenize endpoint](https://docs.basistheory.com/#tokenize) and return the resulting token identifiers to our front end, all without our application having accessed the sensitive card data directly.
```

First, you will need a [Private Application](https://docs.basistheory.com/api-reference/#applications) using our PCI-compliant template `Use PCI Tokens`. This application will be used to securely send cardholder data via the Basis Theory Proxy to the PSP of your choice. [Click here to create one.](https://portal.basistheory.com/applications/create?application_template_id=31efed55-035c-4b49-b1a1-609a728d91ce)

This will create a PCI-compliant application with the following [Access Controls](/concepts/access-controls/):
* Permissions: `token:use`
* Containers: `/pci/`
* Transform: `reveal`

Copy the API Key to be used in the next step.

### Send the Data to the Payment Service Provider

From our backend API, we will now be able to send the cardholder data associated with our token to a third-party PSP.

Add the following code to your API, replacing `PRIVATE_API_KEY` with the API Key from the previous step::

```jsx
const { data }  = await axios.post(
  'https://api.basistheory.com/proxy', 
  {
    card_number: `{%raw%}{{ ${cardTokenId} | json: '$.number' }}{%endraw%}`,
    exp_month: `{%raw%}{{ ${cardTokenId} | json: '$.expiration_month' }}{%endraw%}`,

>     exp_year: `{%raw%}{{ ${cardTokenId} | json: '$.expiration_year' }}{%endraw%}`,

Reviewer: use `to_number` here to show it off?
